packages/devtools_app/lib/src/screens/deep_link_validation/select_project_view.dart

```diff
@@ -0,0 +1,113 @@
+// Copyright 2023 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:devtools_app_shared/ui.dart';
+import 'package:flutter/material.dart';
+
+import '../../shared/config_specific/server/server.dart' as server;
+import '../../shared/directory_picker.dart';
+import '../../shared/utils.dart';
+import 'deep_links_controller.dart';
+import 'deep_links_model.dart';
+
+/// A view for selecting a Flutter project.
+class SelectProjectView extends StatefulWidget {
+  const SelectProjectView({super.key});
+
+  @override
+  State<SelectProjectView> createState() => _SelectProjectViewState();
+}
+
+class _SelectProjectViewState extends State<SelectProjectView>
+    with ProvidedControllerMixin<DeepLinksController, SelectProjectView> {
+  static const _kMessageSize = 24.0;
+  bool _retrievingFlutterProject = false;
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    if (!initController()) return;
+  }
+
+  void _handleDirectoryPicked(String directory) async {
+    setState(() {
+      _retrievingFlutterProject = true;
+    });
+    final List<String> androidVariants =
+        await server.requestAndroidBuildVariants(directory);
+    if (!context.mounted) {
+      return;
+    }
+    if (androidVariants.isEmpty) {
+      await showDialog(
+        context: context,
+        builder: (_) {
+          return AlertDialog(
+            title: const Text('You selected a non Flutter project'),
+            content: const Text(
+              'Seems you selected a non Flutter project. If it is not intended, please reselect a Flutter project',
+            ),
+            actions: [
+              TextButton(
+                child: const Text('Close'),
+                onPressed: () {
+                  Navigator.of(context).pop();
+                },
+              ),
+            ],
+          );
+        },
+      );
+    } else {
+      controller.selectedProject.value =
+          FlutterProject(path: directory, androidVariants: androidVariants);
+    }
+    setState(() {
+      _retrievingFlutterProject = false;
+    });
+  }
+
+  Widget _buildStatusBar(BuildContext context) {
+    Widget? child;
+    if (_retrievingFlutterProject) {
+      child = const SizedBox(
+        width: _kMessageSize,
+        height: _kMessageSize,
+        child: CircularProgressIndicator(),
+      );
```

Comment:
consider using CenteredCircularProgressIndicator for consistency with rest of DevTools